### PR TITLE
Zhalei/mlas test

### PR DESCRIPTION
**Pull request by zhanghuanrong**
Refactor mlas unittest using google test:
    1) sperate into small piece with helper for gtest.
    2) by default run tests from short execute, with --long, could run long run executes.
    3) make heavy short execute test distinguish on different parameter.

### CODE REVIEW

File: cmake/onnxruntime_unittests.cmake [hunk 1]
```diff
@@ -952,14 +952,18 @@ if(WIN32 AND onnxruntime_ENABLE_INSTRUMENT)
   target_link_libraries(compare_two_sessions PRIVATE ${GETOPT_LIB_WIDE} tdh Advapi32)
 endif()
 
-onnxruntime_add_executable(onnxruntime_mlas_test ${TEST_SRC_DIR}/mlas/unittest.cpp)
+file(GLOB onnxruntime_mlas_test_src CONFIGURE_DEPENDS
+  "${TEST_SRC_DIR}/mlas/*.h"
+  "${TEST_SRC_DIR}/mlas/test_*.cpp"
```
Comment (tracysh) [inline cmake/onnxruntime_unittests.cmake:957]: can the code be moved to a subdirectory under test\mlas\unittest, next to bench instead? you wouldn't need the test_* pattern anymore either, just *.cpp.

File: onnxruntime/test/mlas/test_activation.cpp [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            << std::setw(8) << std::setfill('0') <<std::hex << Buffer[i].u << ", expecting:"
+            << std::setw(8) << std::setfill('0') <<std::hex << TestData[i][kind].u;
+      }
+    }
+  }
+};
+
+template<> MlasActivationTest* MlasTestFixture<MlasActivationTest>::mlas_tester(nullptr);
+
+static UNUSED_VARIABLE bool added_to_main = AddTestRegistor([](bool is_short_execute) {
```
Comment (tracysh) [inline onnxruntime/test/mlas/test_activation.cpp:97]: registor->register?

File: onnxruntime/test/mlas/test_conv2d_nchwc.cpp [hunk 3]
```diff
@@ -0,0 +1,36 @@
+
```
Comment (tracysh) [inline onnxruntime/test/mlas/test_conv2d_nchwc.cpp:1]: missing file header

File: onnxruntime/test/mlas/test_main.cpp [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+int main(int argc, char** argv) {
+  bool is_short_execute = (argc <= 1 || strcmp("--long", argv[1]) != 0);
+  std::cout << "-------------------------------------------------------" << std::endl;
+  if (is_short_execute) {
+    std::cout << "----Running quick check mode. Enable more complete test" << std::endl;
+    std::cout << "----  with '--long' as first argument!" << std::endl;
+  }
+  auto test_count = LongShortExecuteManager::instance().RegisterAll(is_short_execute);
+  std::cout << "----Total " << test_count << " tests registered programmablely!" << std::endl;
```
Comment (tracysh) [inline onnxruntime/test/mlas/test_main.cpp:66]: programmablely->programmably

File: cmake/onnxruntime_unittests.cmake (hunk 1 above)
Comment (zhanghuanrong) [reply cmake/onnxruntime_unittests.cmake:957]: no problem, will update it.

File: onnxruntime/test/mlas/test_activation.cpp (hunk 2 above)
Comment (zhanghuanrong) [reply onnxruntime/test/mlas/test_activation.cpp:97]: Will change all.

File: onnxruntime/test/mlas/test_conv2d_nchwc.cpp (hunk 3 above)
Comment (zhanghuanrong) [reply onnxruntime/test/mlas/test_conv2d_nchwc.cpp:1]: Ooops will add it.

File: onnxruntime/test/mlas/test_main.cpp (hunk 4 above)
Comment (zhanghuanrong) [reply onnxruntime/test/mlas/test_main.cpp:66]: my bad English.

**Review by tracysh — APPROVED** [review]

**Pull request merged**